app/src/main/java/org/simple/clinic/protocol/ProtocolRepository.kt

```diff
@@ -163,4 +163,28 @@ class ProtocolRepository @Inject constructor(
     )
   }
 
+  fun dosagesForDrug(drugName: String, protocolUuid: UUID?): Observable<List<String>> {
+    val defaultDosages = defaultProtocolDrugs()
+        .filter { it.drugName == drugName }
+        .flatMap { drugAndDosages -> drugAndDosages.drugs.map { it.dosage } }
+
+    if (protocolUuid == null) {
+      return Observable.just(defaultDosages)
+    }
+
+    val dosages = protocolDrugsDao
+        .dosagesForDrug(drugName, protocolUuid)
+        .replay()
+        .refCount()
+
+    val notEmptyDosages = dosages
+        .filter { it.isNotEmpty() }
+
+    val emptyDosages = dosages
+        .filter { it.isEmpty() }
+        .map { defaultDosages }
```

Comment:
Just like how you've used the names `syncedDrugs` and `defaultDrugs`, can we do the same thing here? That is, use `syncedDosages` and `defaultDosages`?

Reply:
Additionally, the name here does not indicate the data, it indicates what the filter operator is